test/test_http_stress.py

```diff
@@ -0,0 +1,140 @@
+            f = os.path.join(tmp_dir, f"webfile_test_{i}.data")
+
+            write_chunk = 1024 * 1024 * 16
+            rmn_size = file_size
+            while rmn_size > 0:
+                with open(f, "ab+") as fout:
+                    fout.write(os.urandom(min(rmn_size, write_chunk)))
+                rmn_size = rmn_size - min(rmn_size, write_chunk)
+
+            fsum.write(file_url_template.format(num=i))
+
+
+class TestIterableDataPipeHttp(expecttest.TestCase):
+    __server_thread: threading.Thread
+    __server_addr: str
+    __server: socketserver.TCPServer
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            (cls.__server_thread, cls.__server_addr, cls.__server) = set_up_local_server_in_thread()
+        except Exception as e:
+            warnings.warn(
+                "TestIterableDataPipeHttp could\
+                          not set up due to {}".format(
+                    str(e)
+                )
+            )
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cls.__server.shutdown()
+            cls.__server_thread.join(timeout=15)
+        except Exception as e:
+            warnings.warn(
+                "TestIterableDataPipeHttp could\
+                           not tear down (clean up temp directory or terminate\
+                           local server) due to {}".format(
+                    str(e)
+                )
+            )
+
+    def _http_test_base(self, test_file_size, test_file_count, timeout=None, chunk=None):
+        def _get_data_from_tuple_fn(data):
+            return data[1]
+
+        with tempfile.TemporaryDirectory(dir=os.getcwd()) as tmpdir:
+            # create tmp dir and files for test
+            base_tmp_dir = os.path.basename(os.path.normpath(tmpdir))
+            url = "http://{server_addr}/{tmp_dir}/webfile_test_{num}.data\n"
+            file_url_template = url.format(server_addr=self.__server_addr, tmp_dir=base_tmp_dir, num="{num}")
+            create_temp_files_for_serving(tmpdir, test_file_count, test_file_size, file_url_template)
+
+            datapipe_dir_f = FileLister(tmpdir, "*_list")
+            datapipe_stream = FileOpener(datapipe_dir_f, mode="r")
+            datapipe_f_lines = LineReader(datapipe_stream)  # type: ignore[arg-type]
+            datapipe_line_url: IterDataPipe[str] = Mapper(datapipe_f_lines, _get_data_from_tuple_fn)
+            datapipe_http = HttpReader(datapipe_line_url, timeout=timeout)
+            datapipe_tob = StreamReader(datapipe_http, chunk=chunk)
+
+            for (url, data) in datapipe_tob:
+                self.assertGreater(len(url), 0)
+                self.assertRegex(url, r"^http://.+\d+.data$")
+                if chunk is not None:
+                    self.assertEqual(len(data), chunk)
+                else:
+                    self.assertEqual(len(data), test_file_size)
+
+    @unittest.skip("Stress test on large amount of files skipped due to the CI timing constraint.")
```

Comment:
Can you switch it to use `@slowTest`? And, then adding `ciflow/slow` to trigger these tests.

Reply:
I was thinking that if this test truly takes hours to run (as the estimates in the comment indicate), should we have a separate label? Like `@superSlowTest`?

Reply:
Or we can adjust the parameters of the test to make it less slow?
From super slow to acceptable slow, ~1 hour or whatever the max CI can do? There might also be concerns about cost.

Reply:
Could we reduce the time of this stress test to 1 hour-ish? Adding superSlowTest is doable, but I just feel we are not going to really run it with such decorator then this test is going to be useless.

Reply:
Yeah. That's why I think it's good to have one stress test but more admissible for GH CI. Otherwise, I would prefer to delete this test.